spinegeneric/cli/generate_figure.py

```diff
@@ -916,16 +933,222 @@ def main():
+
+        fig = go.Figure()
+        # y_max = height_bar[i_max]+std_sorted[i_max]  # used to display stats
+        # y_max = ax.get_ylim()[1] * 95 / 100  # stat will be located at the top 95% of the graph
+
+        # Generate figure 
+
+        i=0
+        for site in site_sorted:
+            index = list(site_sorted).index(site)
+            val = df['val'][site]
+            val = [value * scaling_factor.get(metric) for value in val]
+            x=site_sorted[i]
+            fig.add_trace(go.Scatter(x=[x,x,x,x,x,x],y=val,mode='markers', marker_color='red',name=x))
+            #fig.update_yaxes(range=[0,max(val)])
+            i=i+1
+        fig.update_traces(marker=dict(size=4))
+
+        fig.add_trace(go.Bar(
+            y=mean_sorted,
+            x=site_sorted,
+            text=model_sorted,
+            textfont=dict(color="white",size=30),
+            textposition="inside",
+            insidetextanchor="start",
+            textangle=-90,
+            error_y=dict(array=std_sorted,
+                        color='#000000',
+                        width=3),
+            marker_color=(list_colors)))
+        
+        # Add stats per vendor
+        x_init_vendor = 0
+
+        for vendor in list(OrderedDict.fromkeys(vendor_sorted)):
+            n_site = list(vendor_sorted).count(vendor)
+            x_i=x_init_vendor - 0.5
+            x_j=x_init_vendor + n_site - 1 + 0.5
+            mean=stats['mean'][vendor]
+            std=stats['std'][vendor]
+            ci=stats['95ci'][vendor]
+            cov_intra=stats['cov_intra'][vendor]
+            cov_inter=stats['cov_inter'][vendor]
+            f=scaling_factor[metric]
+            color=list_colors[x_init_vendor]
+
+            fig.add_trace(go.Scatter(
+                x=[site_sorted[x_init_vendor],site_sorted[x_init_vendor-1+n_site]],
+                y=[mean*f,mean*f],
+                line = dict(color='black', width=1, dash='dash')
+            ))
+
+            fig.add_shape(type="rect",
+                x0=x_i, y0=(mean-std)*f, x1=x_j, y1=(mean+std)*f,
+                line=dict(width=0),
+                opacity=0.2,
+            fillcolor=color)
+
+            #fig.add_annotation(x=x_i, y=max(mean_sorted),
+             #       text= cov_inter * 100)
+
+            x_init_vendor += n_site
+        # try out to add flag on x label
+
+        fig.update_layout(
+            showlegend=False,
+            title='Figure' + ' ' + metric,
+            yaxis_title=metric_to_label_plotly[metric],
+            xaxis_tickangle=-45,
+            bargap=0.4
+        )
+        
+        #Save graph in .html
+        fig.write_html(metric+'.html')
 
         # Get T1w and T2w CSA (will be used later for another figure)
         if metric == "csa_t1":
             csa_t1 = df.sort_values('site').values
         elif metric == "csa_t2":
             csa_t2 = df.sort_values('site').values
 
+    # Sort values per vendor
```

Comment:
some comment for all the code below-- it should be externalized into specific functions